src/Tools/dotnet-monitor/Egress/Extension/EgressExtension.cs

```diff
@@ -148,9 +147,24 @@ public void AddEnvironmentVariable(string key, string value)
 
             parser.BeginReading();
 
-            await JsonSerializer.SerializeAsync<ExtensionEgressPayload>(p.StandardInput.BaseStream, payload, options: null, token);
-            await p.StandardInput.WriteAsync(NewLine, token);
-            await p.StandardInput.BaseStream.FlushAsync(token);
+            // p.StandardInput.BaseStream Format: Version (int), Payload Length (long), Payload, Artifact
+            using Stream intermediateStream = new MemoryStream();
+            await JsonSerializer.SerializeAsync(intermediateStream, payload, options: null, token);
+
+            using (var writer = new BinaryWriter(p.StandardInput.BaseStream, Encoding.UTF8, true))
+            {
+                writer.Write(BitConverter.GetBytes(PayloadProtocolVersion));
```

Comment:
```suggestion
                writer.Write(PayloadProtocolVersion);
```

Since you are not using `BitConverter` on the extension side, then don't use it here. It is fact unnecessary since `BinaryWriter` will write the bytes in a specific endianness and correctly decode it with `BinaryReader`.